elasticdl/python/master/pod_manager.py

```diff
@@ -279,12 +286,29 @@ def _start_worker(self, worker_id):
         worker_args = [self._worker_args[0], job_command]
         envs = copy.deepcopy(self._envs)
         envs.append(V1EnvVar(name=WorkerEnv.WORKER_ID, value=str(worker_id)))
+        need_create_service = False
+        need_patch_service = False
+        ori_index = worker_id
```

Comment:
I think we can maintain a Resource to Pod Map
The Resource contains (priority, service_name ...) which is just like a slot / placeholder, and then we will assign this resource to each created pod. If one pod fails, then the resource is assigned to None and open to a new created pod.
In this way, we can avoid the original_id/new_id logic.